src/lib/mina_state/snarked_ledger_state.ml

```diff
@@ -202,6 +202,8 @@ module Make_str (A : Wire_types.Concrete) = struct
     end
   end]
 
+  type regular_snarked_ledger_state = t
```

Comment:
Is it actually used? If not, can you remove it, please?

In https://github.com/MinaProtocol/mina/commit/bd40afbf7413a82125414cae317aa30a59b88684, I do remove it, in addition to `drop_sok`, and it seems fine.

Reply:
See [here](https://github.com/MinaProtocol/mina/pull/17082)

Reply:
BTW I would also like to drop `regular_snarked_ledger_state` type if possible. Didn't figure it out, yet :(

Reply:
Could you define it like

```ocaml
let drop_sok (t : _ Poly.t) : _ Poly.t = { t with sok_digest = () }
```

That gets you the inferred type

```ocaml
('a, 'b, 'c, 'd, 'e, 'f) Poly.t -> ('a, 'b, 'c, 'd, unit, 'f) Poly.t
```

Reply:
It could go at the top level as well, like the `forget_proofs` in `account_update.ml` that's in https://github.com/MinaProtocol/mina/pull/17012, since it works on any `_ Poly.t`

Reply:
Yes, this make sense :)